src/crypto/CHIPCryptoPALPSA.cpp

```diff
@@ -130,17 +126,31 @@ CHIP_ERROR AES_CCM_encrypt(const uint8_t * plaintext, size_t plaintext_length, c
     status = psa_aead_set_nonce(&operation, nonce, nonce_length);
     VerifyOrReturnError(status == PSA_SUCCESS, CHIP_ERROR_INTERNAL);
 
-    status = psa_aead_update_ad(&operation, aad, aad_length);
-    VerifyOrReturnError(status == PSA_SUCCESS, CHIP_ERROR_INTERNAL);
+    if (aad != nullptr && aad_length != 0)
```

Comment:
```suggestion
    if (aad_length != 0)
```
Pointers are already validated if the corresponding lengths are non-zero.